pkg/mons/restore_quorum.go

```diff
@@ -215,6 +229,24 @@ func updateMonMap(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNa
 	exec.RunCommandInLabeledPod(ctx, clientsets, labelSelector, "mon", "monmaptool", []string{"--print", monmapPath}, clusterNamespace, false, true)
 }
 
+func fetchGoodMon(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace string) string {
+	monStatOutput := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"mon", "stat", "--format", "json-pretty"}, operatorNamespace, clusterNamespace, true, false)
```

Comment:
The scenario for resetting the mon quorum is that quorum has been lost. The challenge here is that `ceph mon stat` will hang if quorum is lost, so it won't do much good to call it if quorum is lost. Where did the suggestion to use `ceph mon stat` come from? I just don't see it helping this scenario.

Instead of `ceph mon stat`, perhaps we should connect to a mon pod and query the admin socket as suggested in this comment:
```
To fetch the leader mon run the command `ceph daemon mon.<name> mon_status` inside mon pod and check for the `state: leader`
```

Reply:
I was working on downstream, and I though, what if happens if someone passes the wrong mon id, it will try to restore based on the wrong mon which is not good mon. Then I was exploring the command and found `ceph mon stat` which has the leader mon. 

do you think we fetch the good mon with `ceph daemon mon.<id> mon_status` command and not depend on the user or this is too much automation?

Reply:
agreed, this way we are validating the monID passed by user and also brings less code change.